pastas/timer.py

```diff
@@ -0,0 +1,37 @@
+try:
+    from tqdm.auto import tqdm
+except ModuleNotFoundError:
+    raise ModuleNotFoundError("SolveTimer requires 'tqdm' to be installed.")
+
+
+class SolveTimer(tqdm):
```

Comment:
I added a solver timeout argument that I apparently haven't pushed yet, I'll quickly add that in.